### ISSUE TITLE
Docker certs 

### ISSUE THREAD

**Issue by lmeyerov** [OWASP/Amass#245]
When I run a command through Docker (`:latest`), I see a bunch of ` x509: certificate signed by unknown authority`: 
 
```
08:00:33.192933 DNSDumpster: The POST request failed: Post https://dnsdumpster.com/: x509: certificate signed by unknown authority
0
...
08:01:53.192937 Data Manager: Gremlin TinkerPop Handler failed to insert infrastructure data: x509: certificate signed by unknown authority
08:01:53.192944 Gremlin TinkerPop Handler failed to insert AAAA record: x509: certificate signed by unknown authority
```

Yet when run directly (via `snap` install) on the same command, it works.

Repros:

Docker
```
docker run -v /home/ubuntu/graphistry/.notebooks/amass/data:/amass/ caffix/amass:latest enum -src -log /amass/zz.err.log
```

Snap
```
 amass enum -config data/config.ini -src -log err.log
```

config.ini:
```
#mode = passive
maximum_dns_queries = 1000

[domains]
domain = graphistry.com

[resolvers]
resolver = 1.1.1.1 ; Cloudflare
resolver = 8.8.8.8 ; Google
resolver = 64.6.64.6 ; Verisign
resolver = 74.82.42.42 ; Hurricane Electric
resolver = 1.0.0.1 ; Cloudflare Secondary
resolver = 8.8.4.4 ; Google Secondary
resolver = 9.9.9.10 ; Quad9 Secondary
resolver = 64.6.65.6 ; Verisign Secondary
resolver = 77.88.8.1 ; Yandex.DNS Secondary

[gremlin]
url = wss://zzzz.gremlin.cosmos.azure.com:443/
username = /dbs/zzz-test-db/colls/zzzz-test-graph
password = yyyy==
```

Some googling suggests this may be some funny docker alpine / go issue, but pretty unclear.

**Comment by VltraHeaven** [comment]
First try passing your config file to the Amass container to see if you get a different result. Since Amass automatically looks for the config file within the database directory specified with the `-dir` flag, you can put your config file in `/home/ubuntu/graphistry/.notebooks/amass/data` on your host and adjust your docker one-liner to look something like this...
```sh
$ cp config.ini /home/ubuntu/graphistry/.notebooks/amass/data
$ docker run -v /home/ubuntu/graphistry/.notebooks/amass/data:/amass/ caffix/amass:latest enum **-dir /amass** -src -log /amass/zz.err.log
```

You can also take it a step further if you think there is something wrong with the container image and use docker to build a fresh container on your host machine.
```sh
$ cd ~/Amass
$ docker build -t amass-local .
$ cp config.ini /home/ubuntu/graphistry/.notebooks/amass/data
$ docker run -v /home/ubuntu/graphistry/.notebooks/amass/data:/amass/ amass-local  enum **-dir /amass** -src -log /amass/zz.err.log
```

**Comment by lmeyerov** [comment]
Pass 1: re:config, yep, it was already there

Pass 2: fresh container, I was using the one on dockerhub, just cleaned and reran:

```
docker ps -a | awk '{ print $1,$2 }' | grep amass | awk '{print $1 }' | xargs -I {} docker rm {}
docker image rm caffix/amass
docker run -v /home/ubuntu/graphistry/.notebooks/amass/data:/amass/ caffix/amass:latest enum -config /amass/config.ini -src -log /amass/zz.err.log
```

Same thing

Happy to try an older tag, tho looks like all from same day on https://hub.docker.com/r/caffix/amass/tags

**Comment by VltraHeaven** [comment]
Ran the following command
```sh
docker run --rm -v $HOME/Dev/go/volume:/Amass --name amass-test caffix/amass:latest enum -dir /Amass -config /Amass/config.ini -src -log /Amass.zz.err.log
```
Got the following results
```sh
[Censys]          api.deis.graphistry.com
[Crtsh]           riverbed.graphistry.com
[Crtsh]           shipyard.graphistry.com
[Crtsh]           fiddlesticks.graphistry.com
[Censys]          est.graphistry.com
[Crtsh]           newlabs.graphistry.com
[Crtsh]           kp.graphistry.com
[ArchiveToday]    internal.deis.graphistry.com
[Crtsh]           311oak833.graphistry.com
[Crtsh]           alpha.graphistry.com
[Crtsh]           estf.graphistry.com
[Forward DNS]     graphistry.com
[Censys]          api.internal.deis.graphistry.com
[Crtsh]           gossip.graphistry.com
[Crtsh]           demo.graphistry.com
[Crtsh]           api.graphistry.com
[Crtsh]           labs.graphistry.com
[Crtsh]           proxy-labs.graphistry.com
[Crtsh]           beta.graphistry.com
[Crtsh]           esto.graphistry.com
[Crtsh]           spk.graphistry.com
[Crtsh]           stuff.graphistry.com
[Censys]          testy.graphistry.com
[Crtsh]           training.graphistry.com
[Crtsh]           lsb.graphistry.com
[Crtsh]           staging.graphistry.com
[Censys]          deis.graphistry.com
[Crtsh]           www.graphistry.com
Average DNS queries performed: 45/sec, DNS names remaining: 3738
[Alterations]     es.graphistry.com
Average DNS queries performed: 65/sec, DNS names remaining: 2812
[Alterations]     qa.graphistry.com
[BufferOver]      proxy-staging.graphistry.com
Average DNS queries performed: 106/sec, DNS names remaining: 1608
[Alterations]     ci.graphistry.com
[BufferOver]      dev.graphistry.com
Average DNS queries performed: 99/sec, DNS names remaining: 969
[HackerTarget]    deploy.graphistry.com
Average DNS queries performed: 119/sec, DNS names remaining: 12
Average DNS queries performed: 46/sec, DNS names remaining: 11
Average DNS queries performed: 12/sec, DNS names remaining: 2
Average DNS queries performed: 1/sec, DNS names remaining: 2
```
Didn't see any certificate errors. A few things to note:
- I noticed you have 'mode = passive' commented out in the config but don't have the opposing 'mode = active' option present.
- You did not show your config specifying which ports to pull certificates from. Maybe these two fields are in your config but you chose not include them in your issue. 
- Amass will automatically save a file named 'indexes.bolt', which saves results and settings from earlier amass enumerations. You can go to `/home/ubuntu/graphistry/.notebooks/amass/data` or just `find ~ -name indexes.bolt`, delete this file and try the enumeration again.

I'm unsure as to why the snap package is functioning yet the docker image is giving you issues. If you want to dig deeper into the issue you can attempt to set a VM/VPS up with docker and try it there to see if docker's virtual network settings, your physical network or your host has an issue that is causing these unknown certificate errors in amass. 

Additionally, (with an awareness that this works fine using the snap version of Amass) If you have completely deleted several options from the config file, it may be beneficial to [re-download the config.ini](https://github.com/OWASP/Amass/blob/master/examples/config.ini), re-insert your API keys and comment out the options you do not need in your enumeration.

Hopefully, some of these suggestions help.

**Comment by caffix** [comment]
Has this problem been fixed for you in Amass v3.1.6?

**Comment by lmeyerov** [comment]
We've been slammed with a few things -- will check next time we're on this stuff (1-2w?), lmk if would be useful to prioritize higher
